### DISCUSSION TITLE
Can I use image compression in the middle of this?

### DISCUSSION THREAD

**Discussion by neciszhang** [donmccurdy/glTF-Transform#1081] [Q&A] [answered]
Hi, I want to use image compression, like tinypng, when converting gltf to glb. This can greatly reduce the model size. I want to know how I should do the conversion in the middle.

**Comment by donmccurdy** [comment]
If you're using the glTF Transform CLI, then the commands below will apply OxiPNG and MozJPEG optimization to PNG and JPEG images. Use both if your model contains both formats.

```bash
gltf-transform png input.gltf output.glb
gltf-transform jpeg output.glb output.glb
```

See `gltf-transform -h` and `gltf-transform png -h` for help.

Alternatively, the same things can be done in a script with the [textureCompress()](https://gltf-transform.dev/modules/functions/functions/textureCompress) function.

**Reply by neciszhang** [reply]
Hi, donmccurdy,
     TextureCompress use sharp,  sharp needs  Node.js environment. i want to use WebIo, can i use textureCompress in  WebIo?

**Reply by donmccurdy** [reply, marked as answer]
Perhaps Sharp will work in a browser someday (partial progress in https://github.com/lovell/sharp/pull/3522) but it doesn't yet. textureCompress() works in a browser in v3.6, just omit the encoder argument, but it is limited to compressing with the browser's own PNG/JPEG codecs. I don't think you'll find that it helps very much, unless the (a)  the original compression on the images was fairly poor, or (b) you're converting to a more efficient format like WebP.

If you are aware of an image compression library you'd like to use instead, then you might be better off just writing that function yourself. See https://github.com/donmccurdy/glTF-Transform/issues/853#issuecomment-1469938223 for an example of how this could be done.
